### DISCUSSION TITLE
Limit login to existing accounts oauth Google only?

### DISCUSSION THREAD

**Discussion by stevenbaert** [open-webui/open-webui#4528] [Q&A]
Hi,

I've enabled OAuth for Open WebUI, and it’s working fine. However, I want to enforce OAuth for existing accounts without allowing new logins. I plan to disable the form-based login, but I'm unsure how to ensure existing users switch to OAuth.

I noticed there's an "OAuth ID" field in Open WebUI that's currently empty. Is there a specific value I need to set for this?Additionally, I've seen the "merge mail" option, but I'm concerned about its security. 

Even with that, I'd still need to allow OAuth account creation, which I want to restrict to a predefined list of email addresses.Any guidance on these issues would be greatly appreciated.

Thanks!
S. 

---

Docs open webui

https://openwebui.com/getting-started/env-configuration/

ENABLE_LOGIN_FORM​

Default: True

Description: Toggles email, password, sign in and "or" (only when ENABLE_OAUTH_SIGNUP is set to true) elements.

Important Note: This should only ever be set to False when ENABLE_OAUTH_SIGNUP is also being used and set to True. Failure to do so will result in the inability to login.

OAUTH_MERGE_ACCOUNTS_BY_EMAIL​

Default: False

Description: If enabled, merges OAuth accounts with existing accounts using the same email address. This is considered unsafe as providers may not verify email addresses and can lead account takeovers.

**Comment by stevenbaert** [comment]
Just a thought: if I do not allow new accounts to be created, create accounts in open-webui upfront like user@gmail.com, then allow only oauth google and set merge accounts .... 

Then only pre-provisoned accounts Gmail can login, right?

**Reply by v1rusnl** [reply]
You need to create and link the google accounts upfront to OWUI. After that you can set ENABLE_OAUTH_SIGNUP=false. So new Oauth login tries with non known Gmail Accounts is not possible anymore and therefor they are visible for the Admin as pending users.

This is at least the way I did it. If you know a better one, please tell me.
